src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/StrandArtifactUnitTest.java

```diff
@@ -0,0 +1,198 @@
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Created by tsato on 4/19/17.
+ */
+public class StrandArtifactUnitTest {
+    final int numChromosomes = 2; // create chromosome 0 and chromosome 1
+    final int startingChromosome = 1;
+    final int chromosomeSize = 1000;
+    final SAMFileHeader SAM_HEADER = ArtificialReadUtils.createArtificialSamHeader(numChromosomes, startingChromosome, chromosomeSize);
+
+    final int chromosomeIndex = 1;
+    final long variantSite = 105;
+    final int alignmentStart = 100;
+    final int readLength = 9;
+
+    final List<Allele> alleles = Arrays.asList(Allele.create((byte) 'C', true), Allele.create((byte) 'A', false));
+    final AlleleList<Allele> alleleList = new IndexedAlleleList<>(alleles);
+    final VariantContext vc = new VariantContextBuilder("source", Integer.toString(chromosomeIndex), variantSite, variantSite, alleles)
+            .attribute(GATKVCFConstants.TUMOR_LOD_KEY, new double[]{ 6.4 }).make();
+
+    final String sampleName = "SamThreetree";
+    final SampleList sampleList = new IndexedSampleList(sampleName);
+
+    final byte baseq = 30;
+    final int mapq = 60;
+
+    // all of the evidence for alt base comes from one strand
+    @Test
+    public void testSNP() throws IOException {
+        // 25% ALT allele fraction at 100x coverage
+        final int numAltReads = 25;
+        final int numRefReads = 75;
+        final int numReads = numAltReads + numRefReads;
+
+        final Map<String, List<GATKRead>> readMap = new LinkedHashMap<>();
+        final List<GATKRead> reads = new ArrayList<>(numReads);
+        final byte[] quals = new byte[readLength];
+        Arrays.fill(quals, baseq);
+
+        // ALT reads are all reverse strand
+        for (int i = 0; i < numAltReads; i++) {
+            final GATKRead read = ArtificialReadUtils.createArtificialRead(SAM_HEADER, "Read" + i, chromosomeIndex, alignmentStart,
+                    "CCCCACCCC".getBytes(), quals, "9M");
+            read.setMappingQuality(mapq);
+            read.setIsReverseStrand(true);
+            reads.add(read);
+        }
+
+        // REF reads have good strand balance
+        for (int i = numAltReads; i < numReads; i++) {
+            final GATKRead read = ArtificialReadUtils.createArtificialRead(SAM_HEADER, "Read" + i, chromosomeIndex, alignmentStart,
+                    "CCCCCCCCC".getBytes(), quals, "9M");
+            read.setMappingQuality(mapq);
+            read.setIsReverseStrand(i % 2 == 0);
+            reads.add(read);
+        }
+        readMap.put(sampleName, reads);
+
+        final ReadLikelihoods<Allele> likelihoods = new ReadLikelihoods<>(sampleList, alleleList, readMap);
+
+        final int sampleIndex = 0;
+        final LikelihoodMatrix<Allele> matrix = likelihoods.sampleMatrix(sampleIndex);
+
+        final double defaultLogLikelihood = 10.0;
```

Comment:
Oh wait, I get it now.  Rename this to something like `logLikelihoodOFBestAllele`.